spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/event/ScopedRefreshRoutesEvent.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.event;
+
+import java.util.Map;
+
+/**
+ * Specialization of {@link RefreshRoutesEvent} to scope refresh by {@link #metadata}.
+ *
+ * @author Ignacio Lozano
+ */
+public class ScopedRefreshRoutesEvent extends RefreshRoutesEvent {
```

Comment:
Is this new class necessary?  Could we just add metadata to the `RefreshRoutesEvent`?

Reply:
We could keep just one. I extended the `RefreshRoutesEvent` class to reduce any impact on other projects using the event and having an immutable event. But we could make 2 constructors though. Let me add a new commit

Reply:
@ryanjbaxter I've just applied the changes amending to the last commit